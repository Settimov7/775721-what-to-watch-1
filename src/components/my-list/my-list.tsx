import * as React from 'react';
import {connect} from 'react-redux';

import {Logo} from "../logo/logo";
import UserBlock from "../user-block/user-block";
import {Footer} from "../footer/footer";

import {getFavoriteFilms} from '../../reducer/films/selectors';
import {Operation} from "../../reducer/films/films";

import {Film} from "../../types";
import {FilmsList} from "../films-list/films-list";

interface Props {
  favoriteFilms: Film[];
  loadFavoriteFilms: () => void;
}

export class MyList extends React.PureComponent<Props, null> {
  componentDidMount(): void {
    this.props.loadFavoriteFilms()
  }

  render() {
    const {favoriteFilms} = this.props;

    return (
      <React.Fragment>
        <div className="user-page">
          <header className="page-header user-page__head">
            <Logo />

            <h1 className="page-title user-page__title">My list</h1>

            <UserBlock />
          </header>

          <section className="catalog">
            <h2 className="catalog__title visually-hidden">Catalog</h2>

            <FilmsList films={favoriteFilms} />
          </section>

          <Footer />
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  favoriteFilms: getFavoriteFilms(state),
});

const mapDispatchToProps = {
  loadFavoriteFilms: Operation.loadFavoriteFilms,
};

export default connect(mapStateToProps, mapDispatchToProps)(MyList);
