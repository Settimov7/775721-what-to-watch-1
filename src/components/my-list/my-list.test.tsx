import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';

import {MyList} from './my-list';

const mockFilms = Array(10).fill(null).map((film, index) => ({
  id: index + 1,
  name: `Film-${index + 1}`,
  posterImageSrc: `/image.jpg`,
  previewImageSrc: `/image.jpg`,
  backgroundImageSrc: `/image.jpg`,
  backgroundColor: `#123456`,
  description: `description`,
  rating: 5,
  scores: 123456,
  director: `director`,
  starring: [`person-1`, `person-2`, `person-3`],
  runTime: 99,
  genre: `Genre`,
  releasedYear: 2019,
  isFavorite: false,
  videoSrc: `video.mp4`,
  previewVideoSrc: `video.mp4`,
}));

it(`My list page correctly renders`, () => {
  const initialMockState = {
    USER: {
      id: 1,
      name: `name`,
      email: `email@mail.com`,
      avatarSrc: `image.src`,
      isAuthorizationRequired: false,
    },
  };
  const mockStore = configureStore();
  const props = {
    favoriteFilms: mockFilms,
    loadFavoriteFilms: jest.fn(),
  };

  const mainPage = renderer.create(
    <Provider store={mockStore(initialMockState)}>
      <BrowserRouter>
        <MyList {...props}/>
      </BrowserRouter>
    </Provider>
  ).toJSON();

  expect(mainPage).toMatchSnapshot();
});
