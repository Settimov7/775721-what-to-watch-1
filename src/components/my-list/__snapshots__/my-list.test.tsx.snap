// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My list page correctly renders 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="image.src"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-1"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/1"
            onClick={[Function]}
          >
            Film-1
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-2"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/2"
            onClick={[Function]}
          >
            Film-2
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-3"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/3"
            onClick={[Function]}
          >
            Film-3
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-4"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/4"
            onClick={[Function]}
          >
            Film-4
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-5"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/5"
            onClick={[Function]}
          >
            Film-5
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-6"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/6"
            onClick={[Function]}
          >
            Film-6
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-7"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/7"
            onClick={[Function]}
          >
            Film-7
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-8"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/8"
            onClick={[Function]}
          >
            Film-8
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-9"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/9"
            onClick={[Function]}
          >
            Film-9
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <img
            alt="Film-10"
            height={175}
            src="/image.jpg"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/10"
            onClick={[Function]}
          >
            Film-10
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
